Logs are not copied in all failure scenarios

http://cstar.datastax.com/tests/id/05524d18-fffd-11e4-8717-42010af0688f

That job obscured this error message, because it didn't copy the logs:

```
INFO  [main] 2015-05-21 14:05:22,588 CassandraDaemon.java:122 - JMX is enabled to receive remote connections on port: 7199
INFO  [main] 2015-05-21 14:05:22,613 CacheService.java:111 - Initializing key cache with capacity of 100 MBs.
INFO  [main] 2015-05-21 14:05:22,624 CacheService.java:133 - Initializing row cache with capacity of 0 MBs
INFO  [main] 2015-05-21 14:05:22,635 CacheService.java:150 - Initializing counter cache with capacity of 50 MBs
INFO  [main] 2015-05-21 14:05:22,638 CacheService.java:161 - Scheduling counter cache save to every 7200 seconds (going to save all keys).
INFO  [main] 2015-05-21 14:05:22,830 ColumnFamilyStore.java:314 - Initializing system.schema_triggers
ERROR [COMMIT-LOG-ALLOCATOR] 2015-05-21 14:05:24,462 CommitLog.java:397 - Failed managing commit log segments. Commit disk failure policy is stop; terminating thread
org.apache.cassandra.io.FSWriteError: java.io.IOException: Invalid argument
        at org.apache.cassandra.db.commitlog.CommitLogSegment.<init>(CommitLogSegment.java:177) ~[main/:na]
        at org.apache.cassandra.db.commitlog.CommitLogSegment.freshSegment(CommitLogSegment.java:119) ~[main/:na]
        at org.apache.cassandra.db.commitlog.CommitLogSegmentManager$1.runMayThrow(CommitLogSegmentManager.java:119) ~[main/:na]
        at org.apache.cassandra.utils.WrappedRunnable.run(WrappedRunnable.java:28) [main/:na]
        at java.lang.Thread.run(Thread.java:745) [na:1.7.0_76]
Caused by: java.io.IOException: Invalid argument
        at java.io.RandomAccessFile.setLength(Native Method) ~[na:1.7.0_76]
        at org.apache.cassandra.db.commitlog.CommitLogSegment.<init>(CommitLogSegment.java:163) ~[main/:na]
        ... 4 common frames omitted

```

--- a/tool/cstar_perf/docker/cstar_docker.py
+++ b/tool/cstar_perf/docker/cstar_docker.py
@@ -56,7 +56,8 @@ RUN \
       ant \
       libjna-java \
       psmisc \
-      python-software-properties
+      python-software-properties \
+      libjpeg-dev
 
 RUN echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | sudo /usr/bin/debconf-set-selections && \
     add-apt-repository ppa:webupd8team/java && \

--- a/tool/cstar_perf/tool/stress_compare.py
+++ b/tool/cstar_perf/tool/stress_compare.py
@@ -150,82 +150,82 @@ def stress_compare(revisions,
             start_fincore_capture(interval=10)
 
         for operation_i, operation in enumerate(operations, 1):
-            start = datetime.datetime.now()
-            stats = {"id":str(uuid.uuid1()), "type":operation['type'],
-                     "revision": revision, "git_id": git_id, "start_date":start.isoformat(),
-                     "label":revision_config.get('label', revision_config['revision'])}
+            try:
+                start = datetime.datetime.now()
+                stats = {"id":str(uuid.uuid1()), "type":operation['type'],
+                         "revision": revision, "git_id": git_id, "start_date":start.isoformat(),
+                         "label":revision_config.get('label', revision_config['revision'])}
 
-            if operation['type'] == 'stress':
-                # Default to all the nodes of the cluster if no
-                # nodes were specified in the command:
-                if operation.has_key('nodes'):
-                    cmd = "{command} -node {hosts}".format(
-                        command=operation['command'],
-                        hosts=",".join(host=operation['nodes']))
-                elif '-node' in operation['command']:
-                    cmd = operation['command']
-                else:
-                    cmd = "{command} -node {hosts}".format(
-                        command=operation['command'],
-                        hosts=",".join([n for n in fab_config['hosts']]))
-                stats['command'] = cmd
-                stats['intervals'] = []
-                stats['test'] = '{operation_i}_{operation}'.format(
-                    operation_i=operation_i, operation=cmd.strip().split(' ')[0]).replace(" ","_")
-                logger.info('Running stress operation : {cmd}  ...'.format(cmd=cmd))
-                # Run stress:
-                # (stress takes the stats as a parameter, and adds
-                #  more as it runs):
-                stress_sha = stress_shas[operation.get('stress_revision', 'default')]
-                stats = stress(cmd, revision, stress_sha, stats=stats)
-                # Wait for all compactions to finish (unless disabled):
-                if operation.get('wait_for_compaction', True):
-                    compaction_throughput = revision_config.get("compaction_throughput_mb_per_sec", 16)
-                    wait_for_compaction(compaction_throughput=compaction_throughput)
+                if operation['type'] == 'stress':
+                    # Default to all the nodes of the cluster if no
+                    # nodes were specified in the command:
+                    if operation.has_key('nodes'):
+                        cmd = "{command} -node {hosts}".format(
+                            command=operation['command'],
+                            hosts=",".join(host=operation['nodes']))
+                    elif '-node' in operation['command']:
+                        cmd = operation['command']
+                    else:
+                        cmd = "{command} -node {hosts}".format(
+                            command=operation['command'],
+                            hosts=",".join([n for n in fab_config['hosts']]))
+                    stats['command'] = cmd
+                    stats['intervals'] = []
+                    stats['test'] = '{operation_i}_{operation}'.format(
+                        operation_i=operation_i, operation=cmd.strip().split(' ')[0]).replace(" ","_")
+                    logger.info('Running stress operation : {cmd}  ...'.format(cmd=cmd))
+                    # Run stress:
+                    # (stress takes the stats as a parameter, and adds
+                    #  more as it runs):
+                    stress_sha = stress_shas[operation.get('stress_revision', 'default')]
+                    stats = stress(cmd, revision, stress_sha, stats=stats)
+                    # Wait for all compactions to finish (unless disabled):
+                    if operation.get('wait_for_compaction', True):
+                        compaction_throughput = revision_config.get("compaction_throughput_mb_per_sec", 16)
+                        wait_for_compaction(compaction_throughput=compaction_throughput)
 
-            elif operation['type'] == 'nodetool':
-                if 'nodes' not in operation:
-                    operation['nodes'] = 'all'
-                if operation['nodes'] in ['all','ALL']:
-                    nodes = [n for n in fab_config['hosts']]
-                else:
-                    nodes = operation['nodes']
+                elif operation['type'] == 'nodetool':
+                    if 'nodes' not in operation:
+                        operation['nodes'] = 'all'
+                    if operation['nodes'] in ['all','ALL']:
+                        nodes = [n for n in fab_config['hosts']]
+                    else:
+                        nodes = operation['nodes']
 
-                set_nodetool_path(os.path.join(product.get_bin_path(), 'nodetool'))
-                logger.info("Running nodetool on {nodes} with command: {command}".format(nodes=operation['nodes'], command=operation['command']))
-                stats['command'] = operation['command']
-                output = nodetool_multi(nodes, operation['command'])
-                stats['output'] = output
-                logger.info("Nodetool command finished on all nodes")
+                    set_nodetool_path(os.path.join(product.get_bin_path(), 'nodetool'))
+                    logger.info("Running nodetool on {nodes} with command: {command}".format(nodes=operation['nodes'], command=operation['command']))
+                    stats['command'] = operation['command']
+                    output = nodetool_multi(nodes, operation['command'])
+                    stats['output'] = output
+                    logger.info("Nodetool command finished on all nodes")
 
-            elif operation['type'] == 'cqlsh':
-                logger.info("Running cqlsh commands on {node}".format(node=operation['node']))
-                set_cqlsh_path(os.path.join(product.get_bin_path(), 'cqlsh'))
-                output = cqlsh(operation['script'], operation['node'])
-                stats['output'] = output.split("\n")
-                logger.info("Cqlsh commands finished")
+                elif operation['type'] == 'cqlsh':
+                    logger.info("Running cqlsh commands on {node}".format(node=operation['node']))
+                    set_cqlsh_path(os.path.join(product.get_bin_path(), 'cqlsh'))
+                    output = cqlsh(operation['script'], operation['node'])
+                    stats['output'] = output.split("\n")
+                    logger.info("Cqlsh commands finished")
 
-            elif operation['type'] == 'bash':
-                nodes = operation.get('nodes', [n for n in fab_config['hosts']])
-                logger.info("Running bash commands on: {nodes}".format(nodes=nodes))
-                stats['output'] = bash(operation['script'], nodes)
-                logger.info("Bash commands finished")
+                elif operation['type'] == 'bash':
+                    nodes = operation.get('nodes', [n for n in fab_config['hosts']])
+                    logger.info("Running bash commands on: {nodes}".format(nodes=nodes))
+                    stats['output'] = bash(operation['script'], nodes)
+                    logger.info("Bash commands finished")
 
-
-            end = datetime.datetime.now()
-            stats['end_date'] = end.isoformat()
-            stats['op_duration'] = str(end - start)
-            log_stats(stats, file=log)
-
-            #Copy node logs:
-            logs_dir = os.path.join(os.path.expanduser('~'),'.cstar_perf','logs')
-            log_dir = os.path.join(logs_dir, stats['id'])
-            os.makedirs(log_dir)
-            retrieve_logs(log_dir)
-            revision_config['last_log'] = stats['id']
-            #Tar them for archiving:
-            subprocess.Popen(shlex.split('tar cfvz {id}.tar.gz {id}'.format(id=stats['id'])), cwd=logs_dir).communicate()
-            shutil.rmtree(log_dir)
+                end = datetime.datetime.now()
+                stats['end_date'] = end.isoformat()
+                stats['op_duration'] = str(end - start)
+                log_stats(stats, file=log)
+            finally:
+                # Copy node logs:
+                logs_dir = os.path.join(os.path.expanduser('~'),'.cstar_perf','logs')
+                log_dir = os.path.join(logs_dir, stats['id'])
+                os.makedirs(log_dir)
+                retrieve_logs(log_dir)
+                revision_config['last_log'] = stats['id']
+                # Tar them for archiving:
+                subprocess.Popen(shlex.split('tar cfvz {id}.tar.gz {id}'.format(id=stats['id'])), cwd=logs_dir).communicate()
+                shutil.rmtree(log_dir)
 
             if capture_fincore:
                 stop_fincore_capture()

